frontend/src/hooks/usePersistentTableState.ts

```diff
@@ -39,7 +39,41 @@ export const usePersistentTableState = <T extends QueryParamConfigMap>(
         queryParamsDefinition,
     );
 
-    const [tableState, setTableState] = useQueryParams(queryParamsDefinition);
+    const [tableState, setTableStateInternal] = useQueryParams(
+        queryParamsDefinition,
+    );
+
+    type SetTableStateInternalParam = Parameters<
+        typeof setTableStateInternal
+    >[0];
+
+    const setTableState = useCallback(
+        (newState: SetTableStateInternalParam) => {
+            if (!queryParamsDefinition.offset) {
+                return setTableStateInternal(newState);
+            }
+            if (typeof newState === 'function') {
+                setTableStateInternal((prevState) => {
+                    const updatedState = (newState as Function)(prevState);
+                    return queryParamsDefinition.offset
+                        ? {
+                              offset: queryParamsDefinition.offset.decode('0'),
+                              ...updatedState,
+                          }
+                        : updatedState;
+                });
+            } else {
+                const updatedState = queryParamsDefinition.offset
+                    ? {
+                          offset: queryParamsDefinition.offset.decode('0'),
```

Comment:
offset is first so that we can overwrite offset for prev/next page in the newState